pkg/chunkenc/unordered_test.go

```diff
@@ -244,37 +246,78 @@ func Test_UnorderedBoundedIter(t *testing.T) {
 }
 
 func TestHeadBlockInterop(t *testing.T) {
-	unordered, ordered := newUnorderedHeadBlock(), &headBlock{}
+	unordered, ordered := newUnorderedHeadBlock(UnorderedHeadBlockFmt), &headBlock{}
+	unorderedWithMetadata := newUnorderedHeadBlock(UnorderedWithMetadataHeadBlockFmt)
 	for i := 0; i < 100; i++ {
-		require.Nil(t, unordered.Append(int64(99-i), fmt.Sprint(99-i)))
-		require.Nil(t, ordered.Append(int64(i), fmt.Sprint(i)))
+		metaLabels := labels.Labels{{Name: "foo", Value: fmt.Sprint(99 - i)}}
+		require.Nil(t, unordered.Append(int64(99-i), fmt.Sprint(99-i), metaLabels))
+		require.Nil(t, unorderedWithMetadata.Append(int64(99-i), fmt.Sprint(99-i), metaLabels))
+		require.Nil(t, ordered.Append(int64(i), fmt.Sprint(i), labels.Labels{{Name: "foo", Value: fmt.Sprint(i)}}))
 	}
 
 	// turn to bytes
-	b1, err := ordered.CheckpointBytes(nil)
+	orderedCheckpointBytes, err := ordered.CheckpointBytes(nil)
 	require.Nil(t, err)
-	b2, err := unordered.CheckpointBytes(nil)
+	unorderedCheckpointBytes, err := unordered.CheckpointBytes(nil)
+	require.Nil(t, err)
+	unorderedWithMetadataCheckpointBytes, err := unorderedWithMetadata.CheckpointBytes(nil)
 	require.Nil(t, err)
 
 	// Ensure we can recover ordered checkpoint into ordered headblock
-	recovered, err := HeadFromCheckpoint(b1, OrderedHeadBlockFmt)
+	recovered, err := HeadFromCheckpoint(orderedCheckpointBytes, OrderedHeadBlockFmt)
 	require.Nil(t, err)
 	require.Equal(t, ordered, recovered)
 
 	// Ensure we can recover ordered checkpoint into unordered headblock
-	recovered, err = HeadFromCheckpoint(b1, UnorderedHeadBlockFmt)
+	recovered, err = HeadFromCheckpoint(orderedCheckpointBytes, UnorderedHeadBlockFmt)
 	require.Nil(t, err)
 	require.Equal(t, unordered, recovered)
 
+	// Ensure we can recover ordered checkpoint into unordered headblock with metadata
+	recovered, err = HeadFromCheckpoint(orderedCheckpointBytes, UnorderedWithMetadataHeadBlockFmt)
+	require.NoError(t, err)
+	require.Equal(t, UnorderedWithMetadataHeadBlockFmt, recovered.Format())
+	require.IsType(t, &unorderedHeadBlock{}, recovered)
+	require.IsType(t, unordered.rt, (recovered.(*unorderedHeadBlock)).rt)
+	require.IsType(t, unordered.size, (recovered.(*unorderedHeadBlock)).size)
+	require.IsType(t, unordered.mint, (recovered.(*unorderedHeadBlock)).mint)
+	require.IsType(t, unordered.maxt, (recovered.(*unorderedHeadBlock)).maxt)
```

Comment:
I think we can do something like this to compare the actual data instead of just types:

```
	require.Equal(t, &unorderedHeadBlock{
		format: UnorderedWithMetadataHeadBlockFmt,
		rt:     unordered.rt,
		lines:  unordered.lines,
		size:   unordered.size,
		mint:   unordered.mint,
		maxt:   unordered.maxt,
	}, recovered)
```

Reply:
Done